cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaReconciler.java

```diff
@@ -915,6 +917,15 @@ protected Future<Void> clusterId(KafkaStatus kafkaStatus) {
                 });
     }
 
+    /**
+     * Configures the default users quota in Kafka in case that the {@link QuotasPluginKafka} is used
+     *
+     * @return  Future which completes when the default quotas are configured
+     */
+    protected Future<Void> defaultKafkaQuotas() {
```

Comment:
so this method is calling something related to the specific Kafka quotas plugin (by using the `DefaultKafkaQuotasManager`) but then internally it's checking if the quota plugin is really an instance of the `QuotasPluginKafka`. I would move the check here and not inside so avoiding the call at all? It would make clear in the reconciliation code that you are going to configure default user quotas but only if the Kafka quotas plugin is used, or?

Reply:
Don't you need to delete the quotas if they are not set in the Kafka CR?

Reply:
Yes as Jakub mentioned, it is there for un-setting the default quotas.
That's for the scenario when we had configuration for the `QuotasPluginKafka` in previous spec, the default quotas were set in Kafka, and now we want to move to `QuotasPluginStrimzi` or `null`. It's not about checking if the actual spec if `QuotasPluginKafka` type or not.

Reply:
My mental model was about expecting a call to a quotas manager which is able to configure the right quotas plugin (if it's Kafka or Strimzi) and not a dedicated call in the reconciler using a dedicated class like the DefaultKafkaQuotasManager which is just about the Kafka plugin (so configuring default quotas or removing them).

Reply:
Strimzi quotas plugin is configured in the broker configuration and this is named `defaultKafkaQuotas`, so it should cover just the Kafka quotas plugin.
From the code I saw that it's both configured on different places and I think it would not be trivial to have something like `configureQuotas()` which would then, based on plugin type, configure the broker configuration for Strimzi plugin or create/alter the default quotas in Kafka using Kafka Admin client.

But maybe I'm wrong